src/test/unit/BLSOperatorStateRetrieverUnit.t.sol

```diff
@@ -0,0 +1,239 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "../utils/MockAVSDeployer.sol";
+
+contract BLSOperatorStateRetrieverUnitTests is MockAVSDeployer {
+    using BN254 for BN254.G1Point;
+
+    uint8 maxQuorumsToRegisterFor = 4;
+    uint256 maxOperatorsToRegister = 4;
+    uint32 registrationBlockNumber = 100;
+    uint32 blocksBetweenRegistrations = 10;
+
+    struct OperatorMetadata {
+        uint256 quorumBitmap;
+        address operator;
+        bytes32 operatorId;
+        BN254.G1Point pubkey;
+        uint96[] stakes; // in every quorum for simplicity
+    }
+
+    function setUp() virtual public {
+        _deployMockEigenLayerAndAVS();
+    }
+
+    function testGetOperatorState_Valid(uint256 pseudoRandomNumber) public {
+        (
+            OperatorMetadata[] memory operatorMetadatas,
+            uint256[][] memory expectedOperatorOverallIndices
+        ) = _registerRandomOperators(pseudoRandomNumber);
+
+        for (uint i = 0; i < operatorMetadatas.length; i++) {
+            uint32 blockNumber = uint32(registrationBlockNumber + blocksBetweenRegistrations * i);
+
+            (uint256 quorumBitmap, BLSOperatorStateRetriever.Operator[][] memory operators) = 
+                operatorStateRetriever.getOperatorState(registryCoordinator, operatorMetadatas[i].operatorId, blockNumber);
+
+            assertEq(operatorMetadatas[i].quorumBitmap, quorumBitmap);
+            bytes memory quorumNumbers = BitmapUtils.bitmapToBytesArray(quorumBitmap);
+            
+            _assertExpectedOperators(
+                quorumNumbers,
+                operators,
+                expectedOperatorOverallIndices,
+                operatorMetadatas
+            );
+        }
+
+        uint256 operatorIndexToDeregister = pseudoRandomNumber % maxOperatorsToRegister;
+        bytes memory quorumNumbersToDeregister = BitmapUtils.bitmapToBytesArray(operatorMetadatas[operatorIndexToDeregister].quorumBitmap);
+        BN254.G1Point memory pubkeyToDeregister = BN254.hashToG1(keccak256(abi.encodePacked("pubkey", pseudoRandomNumber, operatorIndexToDeregister)));
```

Comment:
might want to store this somewhere, or at least have it in a function or something? feels brittle/easy to break to repeatedly calculate this.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/88/commits/8888f2a31ff9e13bca8370846c35f1c48b45d48f